File: Python/Tree_Traversal.py
# inorder: left, root, right
def inorder(root):
    if not root:
        return
    inorder(root.left)
    print(root.val)
    inorder(root.right)

# preorder: root, left, right
def preorder(root):
    if not root:
        return
    print(root.val)
    preorder(root.left)
    preorder(root.right)

# postorder: left, right, root
def postorder(root):
    if not root:
        return
    postorder(root.left)
    postorder(root.right)
    print(root.val)

File: Java/Tree_Traversal.java
class BTreeNode{
    int value;
    BTreeNode left;
    BTreeNode right;

    BTreeNode(int v){
        value = v;
        left = right = null;
    }
}

public class Tree_Traversal {
    public static void preorder(BTreeNode node) {
        if (node==null) {
            return;
        }
        // operation!
        preorder(node.left);
        preorder(node.right);
    }
}